packages/web3-eth-contract/src/index.js

```diff
@@ -180,12 +180,13 @@ var Contract = function Contract(jsonInterface, address, options) {
     // get default account from the Class
     var defaultAccount = this.constructor.defaultAccount;
     var defaultBlock = this.constructor.defaultBlock || 'latest';
-    this.transactionBlockTimeout = this.constructor.transactionBlockTimeout;
-    this.transactionConfirmationBlocks = this.constructor.transactionConfirmationBlocks;
-    this.transactionPollingTimeout = this.constructor.transactionPollingTimeout;
-    this.defaultChain = this.constructor.defaultChain;
-    this.defaultHardfork = this.constructor.defaultHardfork;
-    this.defaultCommon = this.constructor.defaultCommon;
+    this.transactionBlockTimeout = this.options.transactionBlockTimeout || this.constructor.transactionBlockTimeout;
+    this.transactionConfirmationBlocks = this.options.transactionConfirmationBlocks || this.constructor.transactionConfirmationBlocks;
+    this.transactionPollingTimeout = this.options.transactionPollingTimeout || this.constructor.transactionPollingTimeout;
+    this.defaultChain = this.options.defaultChain || this.constructor.defaultChain;
+    this.defaultHardfork = this.options.defaultHardfork || this.constructor.defaultHardfork;
+    this.defaultCommon = this.options.defaultCommon || this.constructor.defaultCommon;
+    this.handleRevert = this.options.handleRevert || this.constructor.handleRevert;
```

Comment:
Is this one a little tricky because it's boolean? For example if I set 
```javascript
web3.eth.handleRevert = true;
```
and later I change my mind for a specific contract
```javascript
web3.eth.Contract.handleRevert = false
```
I think it will still be `true`.

Reply:
The example from you will work and is implemented in the same way [as all other properties](https://github.com/ethereum/web3.js/blob/feature/revert-reason-string/packages/web3-eth/src/index.js#L303). The difference to the ``defaultAccount`` property, for example, is that we are able to define those options over the passed contract constructor options parameter.

Reply:
Oh! Thanks for linking to that part of the code - I misunderstood this.